python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -119,6 +119,27 @@ def __init__(
 
         self._assets_def = check.opt_inst_param(assets_def, "assets_def", AssetsDefinition)
 
+        # These attributes will be set when the context is bound to an op invocation
+        self._op_def = None
+        self._alias = None
+        self._hook_defs = None
+        self._tags = {}
+        self._seen_outputs = {}
+
+        # maintain init time versions of these values so we can unbind the context
+        self._init_op_config = op_config
+        self._init_resources = self._resources
+
+        # Indicates whether the context has been bound to a particular invocation of an op
+        # @op
+        # def my_op(context):
+        #     # context._bound is True
+        #     ...
+        # ctx = build_op_context() # ctx._bound is False
+        # my_op(ctx)
+        # ctx._bound is True, must call ctx.unbind() to unbind
+        self._bound = False
```

Comment:
consider bundling all of these "bound" properties in to their own object - I think the clarity is worth the small hit in having to access through the object `self._bound.XXX`